src/passes/InstrumentMemory.cpp

```diff
@@ -93,14 +94,26 @@ static Name array_set_val_f32("array_set_val_f32");
 static Name array_set_val_f64("array_set_val_f64");
 static Name array_get_index("array_get_index");
 static Name array_set_index("array_set_index");
+static Name memory_grow_pre("memory_grow_pre");
+static Name memory_grow_post("memory_grow_post");
 
 // TODO: Add support for atomicRMW/cmpxchg
 
-struct InstrumentMemory : public WalkerPass<PostWalker<InstrumentMemory>> {
-  // Adds calls to new imports.
-  bool addsEffects() override { return true; }
+using InstructionFilter = std::optional<std::unordered_set<std::string>>;
+
+#define CHECK_EXPRESSION(expr)                                                 \
+  do {                                                                         \
+    if (filter.has_value() && filter->count(expr) == 0) {                      \
```

Comment:
```suggestion
    if (filter && !filter->count(expr)) {                      \
```
Just checking for true/false is enough I believe. And we can use `!x` over `x == 0`.